### PR TITLE
Refactoring InferenceContext and fixing incorrect ctc blank index

### PR DESCRIPTION
**Pull request by ljj7975**
- blank token is meant to be the last label
- Refactored Inference Context

I have tested the change with the trained res8 and seq-lstm models and confirmed that nothing is broken.

I will add updated seq-lstm model later

### CODE REVIEW

File: howl/context.py [hunk 1]
```diff
@@ -24,30 +24,61 @@ def __init__(self,
...
+
         self.coloring = None
-        use_phone = token_type == 'phone'
-        if pronounce_dict is None and use_phone:
-            pronounce_dict = PronunciationDictionary.from_file(SETTINGS.training.phone_dictionary)
-        if use_phone:
+        self.adjusted_vocab = []
+        self.num_labels = 0
+
+        def add_vocab(vocabs: List[str]):
```
Comment (edwinzhng) [inline howl/context.py:32]: Maybe pull this out into a class function instead of nesting it inside __init__?

File: howl/context.py [hunk 2]
```diff
@@ -24,30 +24,61 @@ def __init__(self,
...
-                new_vocab.extend(x.text for x in phone_phrases)
-                self.coloring.extend_sequence(len(phone_phrases))
-            self.coloring.extend_sequence(1)  # negative label
-            if use_blank:
-                self.coloring.append_label(0)  # blank
-            vocab = new_vocab
-            phone_phrases = [PhonePhrase.from_string(x) for x in new_vocab]
+                logging.info(
+                    f'Using phonemes {str(phone_phrases)} for word {word}')
+                add_vocab(x.text for x in phone_phrases)
```
Comment (edwinzhng) [inline howl/context.py:50]: self.add_vocab(...) if the above change is made

**Review by edwinzhng — APPROVED** [review]

**Comment by daemon** [comment]
Can we use 120 columns instead of 80? @ljj7975

**Comment by daemon** [comment]
That's the only part of PEP8 that I (and many others) don't like.

**Pull request merged**